src/librustc_resolve/lib.rs

```diff
@@ -1525,6 +1537,8 @@ impl<'a> Resolver<'a> {
 
         debug!("(resolving item) resolving {}", name);
 
+        self.check_proc_macro_attrs(&item.attrs);
```

Comment:
I believe we also need to call this for items in traits (`ast::TraitItem`) and items in impls (`ast::ImplItem`).